sui-execution/latest/sui-adapter/src/programmable_transactions/execution.rs

```diff
@@ -660,6 +660,25 @@ mod checked {
             invariant_violation!("Tried to normalize modules in existing package but failed")
         };
 
+        let existing_modules_len = current_normalized.len();
+        let upgrading_modules_len = upgrading_modules.len();
+        if context
+            .protocol_config
+            .disallow_new_modules_in_deps_only_packages()
+            && policy as u8 == UpgradePolicy::DEP_ONLY
+            && existing_modules_len != upgrading_modules_len
```

Comment:
Should we add some sort of invariant below about `new_normalized` being empty in the `DEP_ONLY` case if `disallow_new_modules_in_deps_only_packages`

Fine with it being `debug_only`

Reply:
I like it! Will add